lib/collection/src/collection_manager/search_result_aggregator.rs

```diff
@@ -5,6 +5,9 @@ use common::fixed_length_priority_queue::FixedLengthPriorityQueue;
 use common::types::ScoreType;
 use segment::types::{PointIdType, ScoredPoint, SeqNumberType};
 
+/// Avoid excessive memory allocation and allocation failures on huge limits
+const LARGEST_REASONABLE_ALLOCATION_SIZE: usize = 1_048_576;
```

Comment:
I think a lower value should be enough if we want to protect ourselves.

Something like `8,192`